Subject:       Use task_arena when using oneTBB instead of task_scheduler
Patch author:  Orson Teodoro <orsonteodoro@hotmail.com>
Date:          Wed Jun 30 07:15:26 PM PDT 2021 (Unix timestamp: 1625105726)

Copyright notice for cmake/FindTBB.cmake:

//
//   Copyright 2015 Pixar
//
//   Licensed under the Apache License, Version 2.0 (the "Apache License")
//   with the following modification; you may not use this file except in
//   compliance with the Apache License and the following modification to it:
//   Section 6. Trademarks. is deleted and replaced with:
//
//   6. Trademarks. This License does not grant permission to use the trade
//      names, trademarks, service marks, or product names of the Licensor
//      and its affiliates, except as required to comply with Section 4(c) of
//      the License and to reproduce the content of the NOTICE file.
//
//   You may obtain a copy of the Apache License at
//
//       http://www.apache.org/licenses/LICENSE-2.0
//
//   Unless required by applicable law or agreed to in writing, software
//   distributed under the Apache License with the above modification is
//   distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
//   KIND, either express or implied. See the Apache License for the specific
//   language governing permissions and limitations under the Apache License.
//

Examples tested with oneTBB 2021.2.0 in TBB compute mode with
glShareTopology
glEvalLimit

You must pass -DUSE_ONETBB as a CPPFLAG.

diff -urp OpenSubdiv-3_4_3.orig/opensubdiv/osd/tbbEvaluator.cpp OpenSubdiv-3_4_3/opensubdiv/osd/tbbEvaluator.cpp
--- OpenSubdiv-3_4_3.orig/opensubdiv/osd/tbbEvaluator.cpp	2020-04-01 23:04:53.000000000 -0700
+++ OpenSubdiv-3_4_3/opensubdiv/osd/tbbEvaluator.cpp	2021-06-30 19:05:07.642502069 -0700
@@ -22,10 +22,16 @@
 //   language governing permissions and limitations under the Apache License.
 //
 
+// oteodoro - add compatibility for oneTBB
+
 #include "../osd/tbbEvaluator.h"
 #include "../osd/tbbKernel.h"
 
+#ifdef USE_ONETBB						// oteodoro - added line
+#include <oneapi/tbb/task_arena.h>				// oteodoro - added line
+#else								// oteodoro - added line
 #include <tbb/task_scheduler_init.h>
+#endif								// oteodoro - added line
 
 namespace OpenSubdiv {
 namespace OPENSUBDIV_VERSION {
@@ -215,11 +221,19 @@ TbbEvaluator::Synchronize(void *) {
 /* static */
 void
 TbbEvaluator::SetNumThreads(int numThreads) {
+#if defined(USE_ONETBB)						// oteodoro - added line
+    if (numThreads == -1) {					// oteodoro - added line
+        oneapi::tbb::task_arena init;				// oteodoro - added line
+    } else {							// oteodoro - added line
+        oneapi::tbb::task_arena init(numThreads);		// oteodoro - added line
+    }								// oteodoro - added line
+#else								// oteodoro - added line
     if (numThreads == -1) {
         tbb::task_scheduler_init init;
     } else {
         tbb::task_scheduler_init init(numThreads);
     }
+#endif								// oteodoro - added line
 }
 
 }  // end namespace Osd
